module/questions/transformationHelper.js

```diff
@@ -0,0 +1,363 @@
+// Dependencies.
+const { isEmpty, find, capitalize } = require("lodash");
+const { readQuestionSet, readQuestion } = require(ROOT_PATH +
+  "/generics/services/knowledge-platform");
+const { questionType } = require(MODULES_BASE_PATH +
+  "/questions/questionTemplate");
+const { criteriaTemplate, defaultCriteria } = require(MODULES_BASE_PATH +
+  "/criteria/criteriaTemplate");
+const { baseAssessment, assessmentTemplate } = require(MODULES_BASE_PATH +
+  "/assessments/assessmentTemplate");
+
+module.exports = class Transformation {
+  static getQuestionSetHierarchy(
+    submissionDocumentCriterias,
+    solutionDocument,
+    isPageQuestionsRequired = true
+  ) {
+    return new Promise(async (resolve, reject) => {
+      const referenceQuestionSetId = solutionDocument.referenceQuestionSetId;
+
+      const cacheData = await cache
+        .get(`${solutionDocument._id}:${referenceQuestionSetId}`)
+        .catch((err) => {
+          console.log("Error in getting data from redis:", err);
+        });
+
+      if (!!cacheData) {
+        resolve(JSON.parse(cacheData));
+      } else {
+      const res = await readQuestionSet(referenceQuestionSetId).catch((err) => {
+        console.log("Error", err?.response?.data);
+        reject(err?.response?.data);
+      });
+
+      const questionSetHierarchy = res?.result?.questionSet;
+
+      const migratedCriteriaQuestions = questionSetHierarchy?.children || [];
+      const evidences = await this.questionEvidences(
+        migratedCriteriaQuestions,
+        submissionDocumentCriterias,
+        isPageQuestionsRequired
+      );
+
+      assessmentTemplate.assessment.evidences[0].name = capitalize(
+        solutionDocument?.type
+      );
+      assessmentTemplate.assessment.evidences[0].sections[0].name = `${capitalize(
+        solutionDocument?.type
+      )} Questions`;
+
+      assessmentTemplate.assessment.evidences[0].sections[0].questions =
+        evidences?.evidenceSections || [];
+      assessmentTemplate.assessment.evidences[0].sections[0].code = "SQ";
+      assessmentTemplate.assessment.evidences[0].code =
+        assessmentTemplate.assessment.evidences[0].externalId = "SF";
+      assessmentTemplate.assessment.evidences[0].description =
+        questionSetHierarchy?.descripton || "";
+
+      await cache.setEx(
+        `${solutionDocument._id}:${referenceQuestionSetId}`,
```

Comment:
@swetha-aggidevara just solutionId is fine right, why do we need questionSetId as well ?